### PR TITLE
Bug fix in account linking flows

### PR DESCRIPTION
**Pull request by rosalyntan**
Bug fix for an issue where after linking an account with email link, Game Center, or phone auth, the ID token still shows the previous provider (i.e. anonymous), even after force refresh.

[Googler-only internal tracking bug: b/189258684]

### CODE REVIEW

File: FirebaseAuth/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
+# Unreleased
```
Comment (paulb777) [inline FirebaseAuth/CHANGELOG.md:1]: Next release is 8.2.0

**Review by paulb777 — APPROVED** [review]
Does this fix relate to #6521

File: FirebaseAuth/CHANGELOG.md (hunk 1 above)
Comment (rosalyntan) [reply FirebaseAuth/CHANGELOG.md]: Done.

**Comment by rosalyntan** [comment]
> Does this fix relate to #6521

No, I don't believe the issues are related -- in the other issue, the problem is that the ID token and the approximate expiration time are getting out of sync. In this issue, neither the ID token nor the approximate expiration time were being updated properly, and only in account linking flows.

File: FirebaseAuth/Sources/User/FIRUser.m [hunk 2]
```diff
@@ -1104,6 +1113,12 @@ - (void)linkWithCredential:(FIRAuthCredential *)credential
                        if (error) {
                          callInMainThreadWithAuthDataResultAndError(completion, nil, error);
                        } else {
+                         // Update the new token and refresh user info again.
+                         self->_tokenService = [[FIRSecureTokenService alloc]
+                             initWithRequestConfiguration:requestConfiguration
+                                              accessToken:response.IDToken
+                                accessTokenExpirationDate:response.approximateExpirationDate
+                                             refreshToken:response.refreshToken];
```
[the remark this answers is not in the document]
Comment (rosalyntan) [reply FirebaseAuth/Sources/User/FIRUser.m:1121]: Ack.

File: FirebaseAuth/Sources/User/FIRUser.m [hunk 3]
```diff
@@ -1165,6 +1180,12 @@ - (void)linkWithCredential:(FIRAuthCredential *)credential
                           if (error) {
                             callInMainThreadWithAuthDataResultAndError(completion, nil, error);
                           } else {
+                            // Update the new token and refresh user info again.
+                            self->_tokenService = [[FIRSecureTokenService alloc]
+                                initWithRequestConfiguration:requestConfiguration
+                                                 accessToken:response.IDToken
+                                   accessTokenExpirationDate:response.approximateExpirationDate
+                                                refreshToken:response.refreshToken];
```
[the remark this answers is not in the document]
Comment (rosalyntan) [reply FirebaseAuth/Sources/User/FIRUser.m:1188]: Ack.

**Pull request merged**